nibabel/streamlines/tractogram.py

```diff
@@ -0,0 +1,677 @@
+        Each key represents an information $i$ to be kept along side every
+        point of every streamline, and its associated value is an iterable
+        of ndarrays of shape ($N_t$, $M_i$) where $N_t$ is the number of
+        points for a particular streamline $t$ and $M_i$ is the number
+        scalar values to store for that particular information $i$.
+    """
+    def __init__(self, streamlines=None,
+                 data_per_streamline=None,
+                 data_per_point=None,
+                 affine_to_rasmm=np.eye(4)):
+        """
+        Parameters
+        ----------
+        streamlines : iterable of ndarrays or :class:`ArraySequence`, optional
+            Sequence of $T$ streamlines. Each streamline is an ndarray of
+            shape ($N_t$, 3) where $N_t$ is the number of points of
+            streamline $t$.
+        data_per_streamline : dict of iterable of ndarrays, optional
+            Dictionary where the items are (str, iterable).
+            Each key represents an information $i$ to be kept along side every
+            streamline, and its associated value is an iterable of ndarrays of
+            shape ($P_i$,) where $P_i$ is the number scalar values to store
+            for that particular information $i$.
+        data_per_point : dict of iterable of ndarrays, optional
+            Dictionary where the items are (str, iterable).
+            Each key represents an information $i$ to be kept along side every
+            point of every streamline, and its associated value is an iterable
+            of ndarrays of shape ($N_t$, $M_i$) where $N_t$ is the number of
+            points for a particular streamline $t$ and $M_i$ is the number
+            scalar values to store for that particular information $i$.
+        affine_to_rasmm : ndarray of shape (4, 4)
+            Transformation matrix that brings the streamlines contained in
+            this tractogram to *RAS+* and *mm* space where coordinate (0,0,0)
+            refers to the center of the voxel.
+        """
+        self.streamlines = streamlines
+        self.data_per_streamline = data_per_streamline
+        self.data_per_point = data_per_point
+        self._affine_to_rasmm = affine_to_rasmm
+
+    @property
+    def streamlines(self):
+        return self._streamlines
+
+    @streamlines.setter
+    def streamlines(self, value):
+        self._streamlines = ArraySequence(value)
+
+    @property
+    def data_per_streamline(self):
+        return self._data_per_streamline
+
+    @data_per_streamline.setter
+    def data_per_streamline(self, value):
+        self._data_per_streamline = DataPerStreamlineDict(self, value)
+
+    @property
+    def data_per_point(self):
+        return self._data_per_point
+
+    @data_per_point.setter
+    def data_per_point(self, value):
+        self._data_per_point = DataPerPointDict(self, value)
+
+    def get_affine_to_rasmm(self):
```

Comment:
Yes, indeed.  This was why I originally used ``get_affine`` for the image classes.  But, in the end, it seemed too fussy and confusing to prevent something that happens pretty rarely, and that the user may way to do anyway.